File: Emulator/src/main/core/net/event/MSHSWriteEvent.java
package core.net.event;

import core.ServerConstants;
import core.cache.misc.buffer.ByteBufferUtils;
import core.net.IoSession;
import core.net.IoWriteEvent;

import java.nio.ByteBuffer;

/**
 * The type Mshs write event.
 */
public final class MSHSWriteEvent extends IoWriteEvent {

    /**
     * Instantiates a new Mshs write event.
     *
     * @param session the session
     * @param context the context
     */
    public MSHSWriteEvent(IoSession session, Object context) {
        super(session, context);
    }

    @Override
    public void write(IoSession session, Object context) {
        ByteBuffer buffer = ByteBuffer.allocate(2 + ServerConstants.MS_SECRET_KEY.length());
        buffer.put((byte) 88);
        ByteBufferUtils.putString(ServerConstants.MS_SECRET_KEY, buffer);
        session.queue((ByteBuffer) buffer.flip());
    }

}

File: Emulator/src/main/core/net/event/MSHSWriteEvent.kt
package core.net.event

import core.ServerConstants
import core.cache.misc.buffer.ByteBufferUtils
import core.net.IoSession
import core.net.IoWriteEvent
import java.nio.ByteBuffer

/**
 * Handles the management server handshake write event.
 *
 * @author Emperor
 */
class MSHSWriteEvent(session: IoSession, context: Any) : IoWriteEvent(session, context) {

    override fun write(session: IoSession, context: Any) {
        val buffer = ByteBuffer.allocate(2 + ServerConstants.MS_SECRET_KEY.length)
        buffer.put(88.toByte())
        ByteBufferUtils.putString(ServerConstants.MS_SECRET_KEY, buffer)
        buffer.flip()
        session.queue(buffer)
    }
}
